### PR TITLE
Changes to support runtime index based bot response..

### PR DESCRIPTION
**Pull request by sumathi-thirumani**
### Description

Changes to support runtime index based bot response.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- Use-case#1: Successful generation of BOT response with settings "QAzureOpenAIChatConfig".
- Use-case#2: Successful generation of BOT response without settings "QAzureOpenAIChatConfig".
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: webapi/Plugins/Chat/ChatPlugin.cs [hunk 1]
```diff
@@ -546,31 +556,41 @@ private async Task UpdateChatMessageContentAsync(string updatedResponse, string
...
+
         return new OpenAIPromptExecutionSettings
         {
             MaxTokens = this._promptOptions.ResponseTokenLimit,
             Temperature = this._promptOptions.ResponseTemperature,
             TopP = this._promptOptions.ResponseTopP,
             FrequencyPenalty = this._promptOptions.ResponseFrequencyPenalty,
             PresencePenalty = this._promptOptions.ResponsePresencePenalty,
-            ToolCallBehavior = ToolCallBehavior.AutoInvokeKernelFunctions
+            ToolCallBehavior = this._qAzureOpenAIChatExtension.isEnabled() == true ? null : ToolCallBehavior.AutoInvokeKernelFunctions,
```
Comment (wdavidcamden) [inline webapi/Plugins/Chat/ChatPlugin.cs:568]: based on my 10 minutes of reading... should we be using the manual method then?

Comment (wdavidcamden) [reply webapi/Plugins/Chat/ChatPlugin.cs:568]: https://systenics.ai/blog/2024-03-18-function-calling-with-semantic-kernel/ and https://systenics.ai/blog/2024-03-27-manual-function-calling-with-semantic-kernel/

**Comment by sam0044** [comment]
I would, for future, break up this big commit into multiple commits so is easier to follow along.

File: webapi/Plugins/Chat/ChatPlugin.cs (hunk 1 above)
Comment (wdavidcamden) [reply webapi/Plugins/Chat/ChatPlugin.cs:568]: or do we leave it as Auto? or does this stop the ChatExtension from being used?

Comment (sumathi-thirumani) [reply webapi/Plugins/Chat/ChatPlugin.cs]: This value could be either null or Auto for our implementation. Changes done to leave as "Auto". Till the "specialization" feature release; the "meta prompt" in persona needs to be updated with "Role Information".

**Review by wdavidcamden — DISMISSED** [review]

**Pull request closed without merging**